hooks/useMessages.tsx

```diff
@@ -65,14 +70,43 @@ const useMessages = () => {
     messagesRef.current = [];
   };
 
+  const finalCallback = async (
+    message: Message,
+    lastQuestion?: Message,
+    newConversationId?: string,
+    referenceId?: string,
+  ) => {
+    const convId = newConversationId || (pathId as string);
+    const question = lastQuestion || currentQuestion;
+    if (!message.content || !question) return;
+    await trackNewMessage(
+      address as Address,
+      question,
+      selectedArtist?.id || "",
+      convId,
+    );
+    await trackNewMessage(
+      address as Address,
+      {
+        ...message,
+        content: formattedContent(message.content),
+        questionId: question.id,
+      },
+      selectedArtist?.id || "",
+      convId,
+      referenceId,
+    );
+    setCurrentQuestion(null);
+  };
   useEffect(() => {
-    if (messages.length) messagesRef.current = messages;
-  }, [messages]);
-
+    if (messages.length) {
+      messagesRef.current = messages;
+      if (!pending) getPrompts(messages[messages.length - 1]);
+    }
+  }, [messages, tiktokRawReportContent, pending]);
   useEffect(() => {
     if (initialMessages.length) setMessages(initialMessages);
   }, [initialMessages]);
-
   useEffect(() => {
     if (isNewChat) {
       conversationRef.current = "";
```

Comment:
This hook is too long

Dev Requirements: https://zany-hourglass-531.notion.site/Developer-Requirements-Document-15026a7d835c8062bbcde49b59aa00e8

### **Code Standards**

- Maximum file length: 99 lines
- Directory Structure:
    - `/components`: HTML code only
    - `/libs`: Helper functions and API calls
    - `/hooks`: State management and helper function calls
    - `/providers`: Hooks with state shared across multiple components

### **Coding Guidelines**

- Follow principles from "Clean Code" by Robert C. Martin (Uncle Bob)
- Single Responsibility Principle: Each file should perform precisely one function